colossalai/shardformer/layer/attn.py

```diff
@@ -471,19 +471,42 @@ def get_double_ring_groups(sp_group, inner_ring_size=None):
         inner_ring_group = None
         inter_ring_group = None
 
-        # Create inner ring groups
-        for i in range(inner_ring_size):
-            ranks = list(range(i * inner_ring_size, (i + 1) * inner_ring_size))
-            group = dist.new_group(ranks)
-            if sp_rank in ranks:
-                inner_ring_group = group
-
-        # Create inter ring groups
-        for i in range(num_rings):
-            ranks = list(range(i, sp_size, num_rings))
-            group = dist.new_group(ranks)
-            if sp_rank in ranks:
-                inter_ring_group = group
+        world_size = dist.get_world_size()
+        rank = dist.get_rank()
+        groups = int(world_size / sp_size)
```

Comment:
If you just do `world_size / sp_size`, it will loop over all tp * pp * dp ranks?

Reply:
I don't think so, as the tp axis is the first, and sp axis is the next, see as the HybirdPlugin class. groups means the sp groups number. If sp_size = 8, which means there is 8 gpus in one sp group, and if you use two machines to do the trainning(world size = 16),  the groups = 2. So it should be pp * dp ranks.

Reply:
Yeah but sp_group's size doesn't include tp